[Safe CPP] Address Warnings in RenderLayerInlines.h

stwrt opened the pull request:
#### 2b6b1de850dcc46684f8589d40edafd8a04cc673
<pre>
[Safe CPP] Address Warnings in RenderLayerInlines.h
<a href="https://bugs.webkit.org/show_bug.cgi?id=294118">https://bugs.webkit.org/show_bug.cgi?id=294118</a>
<a href="https://rdar.apple.com/problem/152715862">rdar://problem/152715862</a>

Reviewed by Chris Dumez.

Address safe cpp warnings in RenderLayerInlines.h.

 * Source/WebCore/SaferCPPExpectations/UncheckedCallArgsCheckerExpectations:
 * Source/WebCore/SaferCPPExpectations/UncheckedLocalVarsCheckerExpectations:
 * Source/WebCore/rendering/RenderLayer.cpp:
 (WebCore::RenderLayer::paintLayerContents):
 * Source/WebCore/rendering/RenderLayer.h:
 * Source/WebCore/rendering/RenderLayerInlines.h:
 (WebCore::RenderLayer::hasNonOpacityTransparency const):

Canonical link: <a href="https://commits.webkit.org/295950@main">https://commits.webkit.org/295950@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/50b6caebfa7309a174d6b0114f2d2205aee9658a

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/106630 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/26381 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/16778 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/111838 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/57226 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/108669 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/27051 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/34883 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/80976 "Passed tests") | [⏳ 🧪 win-tests](https://ews-build.webkit.org/#/builders/Win-Tests-EWS "Waiting to run tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/109634 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/21455 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/96217 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/61311 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/20908 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/14323 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/56677 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/90794 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/14353 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/114706 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/33768 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/24899 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/90040 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/34132 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/92447 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/89748 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/22914 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/34667 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/12479 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/29395 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/33693 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/39106 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/33439 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/36792 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/35038 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

cdumez commented on Source/WebCore/rendering/RenderLayer.h:
```diff
@@ -180,6 +180,7 @@ class RenderLayer final : public CanMakeSingleThreadWeakPtr<RenderLayer>, public
     inline Page& page() const; // Defined in RenderLayerInlines.h
     inline Ref<Page> protectedPage() const; // Defined in RenderLayerInlines.h
     RenderLayerModelObject& renderer() const { return m_renderer; }
+    CheckedRef<RenderLayerModelObject> checkedRenderer() const { return m_renderer; }
```
Wouldn't the proper fix be to update `m_renderer` to be a `const CheckedRef<RenderLayerModelObject>` ? Then the data member would be safe and we wouldn't need this `checked` getter.

stwrt replied:
Thank you for catching this. I should have set the `m_renderer` to a `const CheckedRef` instead.

cdumez reviewed (approved):
LGTM if the bots are happy.

stwrt committed:
[Safe CPP] Address Warnings in RenderLayerInlines.h
https://bugs.webkit.org/show_bug.cgi?id=294118
rdar://problem/152715862

Reviewed by Chris Dumez.

Address safe cpp warnings in RenderLayerInlines.h.

 * Source/WebCore/SaferCPPExpectations/UncheckedCallArgsCheckerExpectations:
 * Source/WebCore/SaferCPPExpectations/UncheckedLocalVarsCheckerExpectations:
 * Source/WebCore/rendering/RenderLayer.cpp:
 (WebCore::RenderLayer::paintLayerContents):
 * Source/WebCore/rendering/RenderLayer.h:
 * Source/WebCore/rendering/RenderLayerInlines.h:
 (WebCore::RenderLayer::hasNonOpacityTransparency const):

Canonical link: https://commits.webkit.org/295950@main
